source/material_model/steinberger.cc

```diff
@@ -796,6 +692,13 @@ namespace aspect
                                 "prescribed " + Utilities::int_to_string(material_file_names.size()) + " material data files, but there are " +
                                 Utilities::int_to_string(this->n_compositional_fields()) + " compositional fields."));
 
+        if (material_file_names.size() == this->n_compositional_fields() + 1)
```

Comment:
has_background = (material_file_names.size() == this->n_compositional_fields() + 1);